src/main/java/io/fabric8/maven/docker/StopMojo.java

```diff
@@ -105,4 +112,27 @@ private boolean invokedTogetherWithDockerStart() {
         Boolean startCalled = (Boolean) getPluginContext().get(CONTEXT_KEY_START_CALLED);
         return startCalled != null && startCalled;
     }
+
+    private Set<Network> getUnusedCustomNetworks(QueryService queryService, PomLabel pomLabel) throws DockerAccessException {
+        if (!createCustomNetworks) {
+            return Collections.emptySet();
+        }
+        Set<Network> customNetworks = new HashSet<>();
+        for (ImageConfiguration image : getImages()) {
+            final NetworkingMode networkingMode = image.getRunConfiguration().getNetworkingMode();
+            if (networkingMode.isCustomNetwork()) {
+                Network network = queryService.getNetworkByName(networkingMode.getCustomNetwork());
```

Comment:
Probably a `Set<Network> customNetworks = queryService.getNetworksByName()` is better here since you then need only a single HTTP request to the Docker daemon instead of one for each image.